examples/silentpayments.c

```diff
@@ -0,0 +1,543 @@
+                 *  cache of every label he has previously used and uses a
+                 *  callback to check if a potential label exists in his cache.
+                 *  Since the labels are created using an incremental integer
+                 *  `m`, if Bob ever forgets how many labels he has previously
+                 *  used, he can pregenerate a large number of labels e.g.,
+                 *  0..100_000.
+                 */
+                size_t len = 33;
+                secp256k1_pubkey label;
+                unsigned int m = 1;
+
+                /* Load Bob's spend public key */
+                ret = secp256k1_ec_pubkey_parse(ctx,
+                    &spend_pubkey,
+                    bob_spend_pubkey,
+                    33
+                );
+
+                /* Add an entry to the cache. This implies Bob has previously
+                 * called
+                 * `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey`
+                 * and is using the resulting labelled spend pubkey to encode a
+                 * labelled silent payments address.
+                 */
+                ret &= secp256k1_silentpayments_recipient_create_label(ctx,
+                    &label,
+                    labels_cache.entries[0].label_tweak,
+                    bob_scan_key,
+                    m
+                );
+                ret &= secp256k1_ec_pubkey_serialize(ctx,
+                    labels_cache.entries[0].label,
+                    &len,
+                    &label,
+                    SECP256K1_EC_COMPRESSED
+                );
+                assert(ret);
+                labels_cache.entries_used = 1;
+            }
+
+            /* Bob collects the public data from the transaction inputs and
+             * creates a `secp256k1_silentpayments_recipient_public_data` object. He uses
+             * this for his own scanning and also serializes the `public_data`
+             * object to send to light clients. We will use this later for
+             * Carol, who is scanning as a light client. Note, anyone can create
+             * and provide these `public_data` objects, i.e. you don't need to be
+             * a silent payments wallet, just someone interested in providing this
+             * data to light clients, e.g. a wallet service provider. In our
+             * example, Bob is scanning for himself but also sharing this data
+             * with light clients.
+             */
+            ret = secp256k1_silentpayments_recipient_public_data_create(ctx,
+                &public_data,
+                smallest_outpoint,
+                tx_input_ptrs, N_INPUTS,
+                NULL, 0 /* NULL because no eligible plain pubkey inputs were found in the tx */
+            );
+            ret &= secp256k1_silentpayments_recipient_public_data_serialize(ctx,
+                light_client_data33,
+                &public_data
```

Comment:
We should check `ret` after `public_data_create` and before potentially using `data_serialize` with invalid values.

Reply:
Done.